### PR TITLE
Relation update

### PR DESCRIPTION
**Pull request by emma4you**
Pull request ref #25975

@alexef @anaion @iuliachiriac

Please review and merge

### CODE REVIEW

File: hstool/templates/tool/relation_add.html [hunk 1]
```diff
@@ -43,172 +64,190 @@
...
+		<div class="col-sm-offset-2">
+		  <button type="submit" name="draft" value="false"
+				  class="btn btn-primary">Publish</button>
+		  <button type="submit" name="draft" value="true"
+				  class="btn btn-default">Save as draft</button>
+		  <a href="{% url 'assessments:detail' assessment.id %}"
+			 class="btn btn-link" role="button">Cancel</a>
+		</div>
+	  </div>
+	</form>
```
Comment (iuliachiriac) [inline hstool/templates/tool/relation_add.html]: Fix indentation. You should use 2 spaces indentation for html/jinja templates. Also for javascript files.

File: hstool/forms.py [hunk 2]
```diff
@@ -107,6 +112,10 @@ def clean(self):
             self.cleaned_data['destination'] = self.cleaned_data['driver']
         elif self.cleaned_data['figureindicator']:
             self.cleaned_data['destination'] = self.cleaned_data['figureindicator']
+            self.cleaned_data['description'] = ''
```
Comment (anaion) [inline hstool/forms.py]: Why not handle description clearing in the JS as in the other cases?

File: hstool/templates/tool/relation_add.html [hunk 3]
```diff
@@ -6,34 +6,54 @@
...
+          $('#id_figureindicator').prop('selectedIndex', 0);
+          $('#id_impact').prop('selectedIndex', 0);
           }
-          else if (figureindicator_checkbox.checked) {
-              $('[data-show-if=show-figureindicators]').removeClass('hide');
-              $('[data-show-if=show-drivers]').addClass('hide');
-              $('#id_driver').prop('selectedIndex', 0);
-          }
-          else {
+        else if (figureindicator_checkbox.checked) {
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: I think this JS section is starting to be difficult to be managed :(. Can you please move the reused code to separate functions and call them for each check selector?

File: hstool/templates/tool/relation_add.html [hunk 4]
```diff
@@ -4,36 +4,60 @@
 
 {% block scripts %}
   <script>
+    function add_hide_all() {
+     $('[data-show-if=show-impacts]').addClass('hide');
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: Please fix indentation.

File: hstool/templates/tool/relation_add.html [hunk 5]
```diff
@@ -4,36 +4,60 @@
...
+     $('[data-show-if=show-figureindicators]').addClass('hide');
+     $('[data-show-if=show-drivers]').addClass('hide');
+    }
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: One empty line between functions

File: hstool/templates/tool/relation_add.html [hunk 6]
```diff
@@ -4,36 +4,60 @@
...
+        var driver_checkbox = document.getElementById("id_check_driver");
+        var figureindicator_checkbox = document.getElementById("id_check_figureindicator");
+        var impact_checkbox = document.getElementById("id_check_impact");
+
+        $(':checkbox').not(this).prop('checked', false);
+        add_hide_all();
+        if (driver_checkbox.checked) {
+          $('[data-show-if=show-drivers]').removeClass('hide');
+          deselect('driver');
           }
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: Please align brackets

File: hstool/templates/tool/relation_add.html [hunk 7]
```diff
@@ -4,36 +4,60 @@
...
-          else if (figureindicator_checkbox.checked) {
-              $('[data-show-if=show-figureindicators]').removeClass('hide');
-              $('[data-show-if=show-drivers]').addClass('hide');
-              $('#id_driver').prop('selectedIndex', 0);
+        if (figureindicator_checkbox.checked) {
+          $('[data-show-if=show-figureindicators]').removeClass('hide');
+          deselect('figureindicator');
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: Indentation

File: hstool/templates/tool/relation_add.html [hunk 8]
```diff
@@ -4,36 +4,60 @@
...
-        document.getElementById("id_check_figureindicator").checked = true;
-        $('[data-show-if=show-figureindicators]').removeClass('hide');
+          document.getElementById("id_check_figureindicator").checked = true;
+          $('[data-show-if=show-figureindicators]').removeClass('hide');
       }
 
       if ({{ show_driver_of_change_check }}) {
-        document.getElementById("id_check_driver").checked = true;
-        $('[data-show-if=show-drivers]').removeClass('hide');
+          document.getElementById("id_check_driver").checked = true;
```
Comment (anaion) [inline hstool/templates/tool/relation_add.html]: Here the lines are 4 spaces indented.

**Comment by anaion** [comment]
Ok to merge after conding style fixes.

**Pull request merged**